src/CMakeLists.txt

```diff
@@ -360,6 +360,9 @@ add_custom_target(HalideIncludes ALL
 # Define the Halide library target.
 ##
 
+set(CXX_COMPILER_IS_GNU_OR_CLANG $<CXX_COMPILER_ID:GNU,Clang,AppleClang>)
+set(CXX_COMPILER_IS_GNU_OR_CLANG_AND_NOT_CLANGCL $<AND:${CXX_COMPILER_IS_GNU_OR_CLANG},$<NOT:$<STREQUAL:${CMAKE_CXX_SIMULATE_ID},MSVC>>>)
```

Comment:
insert sad comment here about why this long identifier actually is necessary

Reply:
It could perhaps be `CXX_IS_GNULIKE` and `CXX_IS_GNULIKE_NOT_CLANGCL`

Reply:
> `CXX_IS_GNULIKE`

did you mean: `CXX_IS_GNU_SLASH_LINUX_LIKE`

:-)